path_login.go

```diff
@@ -88,129 +86,34 @@ func (b *jwtAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
-			}
-			if err := json.Unmarshal(payload, &allClaims); err != nil {
-				return nil, fmt.Errorf("failed to unmarshal claims: %v", err)
-			}
-		} else {
-			parsedJWT, err := jwt.ParseSigned(token)
-			if err != nil {
-				return logical.ErrorResponse(errwrap.Wrapf("error parsing token: {{err}}", err).Error()), nil
-			}
-
-			var valid bool
-			for _, key := range config.ParsedJWTPubKeys {
-				if err := parsedJWT.Claims(key, &claims, &allClaims); err == nil {
-					valid = true
-					break
-				}
-			}
-			if !valid {
-				return logical.ErrorResponse("no known key successfully validated the token signature"), nil
-			}
-		}
-
-		// We require notbefore or expiry; if only one is provided, we allow 5 minutes of leeway by default.
-		// Configurable by ExpirationLeeway and NotBeforeLeeway
-		if claims.IssuedAt == nil {
-			claims.IssuedAt = new(jwt.NumericDate)
-		}
-		if claims.Expiry == nil {
-			claims.Expiry = new(jwt.NumericDate)
-		}
-		if claims.NotBefore == nil {
-			claims.NotBefore = new(jwt.NumericDate)
-		}
-		if *claims.IssuedAt == 0 && *claims.Expiry == 0 && *claims.NotBefore == 0 {
-			return logical.ErrorResponse("no issue time, notbefore, or expiration time encoded in token"), nil
-		}
-
-		if *claims.Expiry == 0 {
-			latestStart := *claims.IssuedAt
-			if *claims.NotBefore > *claims.IssuedAt {
-				latestStart = *claims.NotBefore
-			}
-			leeway := role.ExpirationLeeway.Seconds()
-			if role.ExpirationLeeway.Seconds() < 0 {
-				leeway = 0
-			} else if role.ExpirationLeeway.Seconds() == 0 {
-				leeway = claimDefaultLeeway
-			}
-			*claims.Expiry = jwt.NumericDate(int64(latestStart) + int64(leeway))
-		}
-
-		if *claims.NotBefore == 0 {
-			if *claims.IssuedAt != 0 {
-				*claims.NotBefore = *claims.IssuedAt
-			} else {
-				leeway := role.NotBeforeLeeway.Seconds()
-				if role.NotBeforeLeeway.Seconds() < 0 {
-					leeway = 0
-				} else if role.NotBeforeLeeway.Seconds() == 0 {
-					leeway = claimDefaultLeeway
-				}
-				*claims.NotBefore = jwt.NumericDate(int64(*claims.Expiry) - int64(leeway))
-			}
-		}
-
-		if len(claims.Audience) > 0 && len(role.BoundAudiences) == 0 {
-			return logical.ErrorResponse("audience claim found in JWT but no audiences bound to the role"), nil
-		}
-
-		expected := jwt.Expected{
-			Issuer:  config.BoundIssuer,
-			Subject: role.BoundSubject,
-			Time:    time.Now(),
-		}
-
-		cksLeeway := role.ClockSkewLeeway
-		if role.ClockSkewLeeway.Seconds() < 0 {
-			cksLeeway = 0
-		} else if role.ClockSkewLeeway.Seconds() == 0 {
-			cksLeeway = jwt.DefaultLeeway
-		}
-
-		if err := claims.ValidateWithLeeway(expected, cksLeeway); err != nil {
-			return logical.ErrorResponse(errwrap.Wrapf("error validating claims: {{err}}", err).Error()), nil
-		}
+	// Get the JWT validator based on the configured auth type
+	validator, err := b.jwtValidator(config)
+	if err != nil {
+		return logical.ErrorResponse(errwrap.Wrapf("error configuring token validator: {{err}}", err).Error()), nil
```

Comment:
I know the pattern predate this text, but I'm not seeing the need for wrapping since the resulting parameter is just a string.  Is this the same?
```
	return logical.ErrorResponse("error configuring token validator: %s", err.Error()), nil
```

Reply:
Yep, the wrapping is not needed. The formatted string you provided results in the same error response. Done in a9828c5.